File: maxProduct.c


int maxProduct(int* nums, int numsSize){
    if(numsSize==1)return (nums[0]-1);
    int max=0;
    int product=0;
    for(int i = 0; i<numsSize; i++){
        for(int j = i+1; j<numsSize; j++){
            product=(nums[i]-1)*(nums[j]-1);
            if(product>max){
                max=product;
            }
        }
    }
    return max;
}

File: maxProduct.java
class Solution {
    public int maxProduct(int[] nums) {
        int curr_max=0;
        int curr_min=0;
        int result=0;
        
        for(int i = 0; i<nums.length; i++){
            int product = curr_max;
            
            curr_max=Math.max(Math.max(curr_max*nums[i],curr_min*nums[i]),nums[i]);
            curr_min=Math.min(Math.min(product*nums[i],curr_min*nums[i]),nums[i]);
            
            if(curr_max>result){
                result=curr_max;
            }    
        }
        return result;
    }
}